### PR TITLE
feat(dato-panel): add is open observable to the service

### PR DESCRIPTION
**Pull request by shaharkazaz**

### CODE REVIEW

File: src/app/components/panel/examples/panel.ex.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 import { DatoPanel } from '../../../../../projects/datorama/core/src/lib/panel/panel.service';
 
 @Component({
   template: `
-    <button class="mr-10 action-btn" (click)="open()">Open panel</button> <button class="mr-10 action-btn" (click)="openAnother()">Open different panel</button> <button class="mr-10 action-btn" (click)="openBackdrop()">Open panel with backdrop</button>
-
+    <button class="mr-10 action-btn" (click)="open()">Open panel</button> <button class="mr-10 action-btn" (click)="openBackdrop()">Open panel with backdrop</button> <button class="mr-10 action-btn" [disabled]="!isOpen" (click)="closeWithData()">Close with data</button>
     <section>
-      <nav>Nav</nav>
+      <nav datoColor="primary-700">Panel state: {{ (isOpen$ | async) ? 'open' : 'close' }}</nav>
```
Comment (NetanelBasal) [inline src/app/components/panel/examples/panel.ex.ts]:
```suggestion
      <nav datoColor="primary-700">Panel state: {{ (isOpen$ | async) ? 'Open' : 'Close' }}</nav>
```

Comment (shaharkazaz) [reply src/app/components/panel/examples/panel.ex.ts]: 👍

File: src/app/components/panel/panel-docs.component.ts [hunk 2]
```diff
@@ -2,23 +2,17 @@ import { Component } from '@angular/core';
...
 import { PanelExampleComponent } from './examples/panel.ex';
 import PanelExampleCode from '!!raw-loader!./examples/panel.ex';
-import { PanelCustomRelativeExampleComponent } from './examples/panel-custom-relative.ex';
-import PanelCustomRelativeExampleCode from '!!raw-loader!./examples/panel-custom-relative.ex';
 
 @Component({
   template: `
     <core-example [code]="PanelExampleCode" [component]="PanelExampleComponent" isColumn="true"></core-example>
 
-    <core-example isColumn="true" title="Custom Relative" [code]="PanelCustomRelativeExampleCode" [component]="PanelCustomRelativeExampleComponent"></core-example>
```
Comment (NetanelBasal) [inline src/app/components/panel/panel-docs.component.ts]: Why you removed this example?

Comment (shaharkazaz) [reply src/app/components/panel/panel-docs.component.ts]: no need for 2 big examples I merged them into 1, part of the doc refactor there

File: projects/datorama/core/src/lib/panel/panel.service.ts [hunk 3]
```diff
@@ -31,11 +31,13 @@ export class DatoPanel {
   private backDropHolder = 'body';
 
   private panelsStack: DatoPanelInternalConfig[] = [];
-
+  private _isOpen = new BehaviorSubject<boolean>(false);
   private get panelContainer() {
     return this.document.querySelector('.dato-panel-container');
   }
 
+  isOpen$ = this._isOpen.asObservable().pipe(distinctUntilChanged());
```
Comment (NetanelBasal) [inline projects/datorama/core/src/lib/panel/panel.service.ts]: What is the motivation for this feature?

Comment (shaharkazaz) [reply projects/datorama/core/src/lib/panel/panel.service.ts]: I need to get an update when the panel status changes

File: src/app/components/panel/panel-docs.component.ts [hunk 4]
```diff
@@ -2,23 +2,17 @@ import { Component } from '@angular/core';
 import api from './api';
 import { PanelExampleComponent } from './examples/panel.ex';
 import PanelExampleCode from '!!raw-loader!./examples/panel.ex';
-import { PanelCustomRelativeExampleComponent } from './examples/panel-custom-relative.ex';
-import PanelCustomRelativeExampleCode from '!!raw-loader!./examples/panel-custom-relative.ex';
 
 @Component({
   template: `
     <core-example [code]="PanelExampleCode" [component]="PanelExampleComponent" isColumn="true"></core-example>
```
[the remark this answers is not in the document]
Comment (shaharkazaz) [reply src/app/components/panel/panel-docs.component.ts]: This week

File: src/app/components/panel/examples/panel.ex.ts [hunk 5]
```diff
@@ -1,16 +1,17 @@
 import { Component, OnDestroy } from '@angular/core';
-import { PanelDemoTwoComponent } from './panel-demo-two.component';
 import { PanelDemoComponent } from './panel-demo.component';
 import { DatoPanel } from '../../../../../projects/datorama/core/src/lib/panel/panel.service';
 
 @Component({
   template: `
-    <button class="mr-10 action-btn" (click)="open()">Open panel</button> <button class="mr-10 action-btn" (click)="openAnother()">Open different panel</button> <button class="mr-10 action-btn" (click)="openBackdrop()">Open panel with backdrop</button>
-
+    <button class="mr-10 action-btn" (click)="open()">Open panel</button> <button class="mr-10 action-btn" (click)="openBackdrop()">Open panel with backdrop</button> <button class="mr-10 action-btn" [disabled]="!isOpen" (click)="closeWithData()">Close with data</button>
```
Comment (NetanelBasal) [inline src/app/components/panel/examples/panel.ex.ts]: button in new line

File: src/app/components/panel/examples/panel.ex.ts [hunk 6]
```diff
@@ -43,11 +55,11 @@ export class PanelExampleComponent implements OnDestroy {
...
       })
       .afterClosed()
-      .subscribe(() => console.log('closed'));
+      .subscribe(res => console.log('closed', res));
   }
 
-  close() {
-    this.panelService.close();
+  closeWithData() {
+    this.panelService.close('Data Passed');
```
Comment (NetanelBasal) [inline src/app/components/panel/examples/panel.ex.ts]: make it more realistic and pass a dummy object.

File: src/app/components/panel/examples/panel.ex.ts (hunk 5 above)
Comment (shaharkazaz) [reply src/app/components/panel/examples/panel.ex.ts]: not sure I understand

File: src/app/components/panel/examples/panel.ex.ts (hunk 6 above)
Comment (shaharkazaz) [reply src/app/components/panel/examples/panel.ex.ts]: ok

File: src/app/components/panel/examples/panel.ex.ts (hunk 5 above)
Comment (NetanelBasal) [reply src/app/components/panel/examples/panel.ex.ts]: all the buttons are in one line (the code).

Comment (shaharkazaz) [reply src/app/components/panel/examples/panel.ex.ts]: it's the prettier, I put them each in a sperate line.
look at the code before, it was one line as well

File: src/app/components/panel/examples/panel.ex.ts [hunk 7]
```diff
@@ -59,7 +59,7 @@ export class PanelExampleComponent implements OnDestroy {
   }
 
   closeWithData() {
-    this.panelService.close('Data Passed');
+    this.panelService.close({ id: 123, otherProp: true });
```
Comment (NetanelBasal) [inline src/app/components/panel/examples/panel.ex.ts]: next time use `git commit --amend` as I wrote in the docs.

Comment (shaharkazaz) [reply src/app/components/panel/examples/panel.ex.ts]: sure.
which docs?

Comment (NetanelBasal) [reply src/app/components/panel/examples/panel.ex.ts]: http://dacorama.surge.sh/documentation/start

**Pull request merged**